students/rmart300/mailroom/mailroom.py

```diff
@@ -44,22 +51,18 @@ def create_report():
     donor_list = ['Fred Smith','Terrie Ann','Murray Martin','Josh Jones','Jane Doe']
     amount_list = [500,100,1000]
     for donor in donor_list:
-        if donor not in donation_dict:
-            donation_dict[donor]=[]
+        donation_dict[donor]=amount_list
     
-        for amount in amount_list:
-            donation_list = donation_dict[donor] 
-            donation_list.append(amount)
-            donation_dict[donor] = donation_list
 
     #prompt user for action and then call function
-    action = ''
-    while action != 'quit':
-        action = input("What would you like to do: \“Send a Thank You\”, \“Create a Report\” or \“quit\”\n")
+    action = 0
+    while not is_number(action) or int(action) != 3:
+        action = input('What would you like to do: 1 - \“Send a Thank You\”, 2 - \“Create a Report\” or 3 - \“quit\”\n')
         
-        if action.lower() == "send a thank you":
+        if not is_number(action) or int(action) < 1 or int(action) > 3:
+            print("please enter a number 1-3")
+        elif int(action) == 1:
             send_thank_you()
-        elif action.lower() == "create a report":
-            print("got here")
+        elif int(action) == 2:

```

Comment:
I think we can make this block even cleaner in the next portion of mailroom where we'll get to use try/except syntax. so keep that in mind for next assignment 😉